Source/WebCore/dom/Element.cpp

```diff
@@ -2603,6 +2614,12 @@ Node::InsertedIntoAncestorResult Element::insertedIntoAncestor(InsertionType ins
             ensureElementRareData().setEffectiveLang(lang);
     }
 
+    if (!UNLIKELY(hasRareData()) || !elementRareData()->effectiveLang()) {
+        auto langAttr = langFromAttribute();
```

Comment:
Again, this langFromAttribute call will do O(k) traversal where k is the number of attributes on this element so we really want to avoid that by using a bit on EventTarget.